pkg/scheduler/factory.go

```diff
@@ -18,182 +18,81 @@ package scheduler
+		prof := &profiles[i]
+		var found = false
+		for k := range prof.PluginConfig {
+			if prof.PluginConfig[k].Name == noderesources.Name {
+				// Update the existing args
+				pc := &prof.PluginConfig[k]
+				args, ok := pc.Args.(*schedulerapi.NodeResourcesFitArgs)
+				if !ok {
+					return nil, fmt.Errorf("want args to be of type NodeResourcesFitArgs, got %T", pc.Args)
 				}
-			}
-			if !found {
-				return nil, fmt.Errorf("can't find NodeResourcesFitArgs in plugin config")
+				args.IgnoredResources = ignoredExtendedResources
+				found = true
+				break
 			}
 		}
+		if !found {
+			return nil, fmt.Errorf("can't find NodeResourcesFitArgs in plugin config")
+		}
 	}
-
-	// The nominator will be passed all the way to framework instantiation.
-	nominator := internalqueue.NewPodNominator(c.informerFactory.Core().V1().Pods().Lister())
-	profiles, err := profile.NewMap(c.profiles, c.registry, c.recorderFactory,
-		frameworkruntime.WithComponentConfigVersion(c.componentConfigVersion),
-		frameworkruntime.WithClientSet(c.client),
-		frameworkruntime.WithKubeConfig(c.kubeConfig),
-		frameworkruntime.WithInformerFactory(c.informerFactory),
-		frameworkruntime.WithSnapshotSharedLister(c.nodeInfoSnapshot),
-		frameworkruntime.WithRunAllFilters(c.alwaysCheckAllPredicates),
-		frameworkruntime.WithPodNominator(nominator),
-		frameworkruntime.WithCaptureProfile(frameworkruntime.CaptureProfile(c.frameworkCapturer)),
-		frameworkruntime.WithClusterEventMap(c.clusterEventMap),
-		frameworkruntime.WithParallelism(int(c.parallellism)),
-		frameworkruntime.WithExtenders(extenders),
-	)
-	if err != nil {
-		return nil, fmt.Errorf("initializing profiles: %v", err)
-	}
-	if len(profiles) == 0 {
-		return nil, errors.New("at least one profile is required")
-	}
-	// Profiles are required to have equivalent queue sort plugins.
-	lessFn := profiles[c.profiles[0].SchedulerName].QueueSortFunc()
-	podQueue := internalqueue.NewSchedulingQueue(
-		lessFn,
-		c.informerFactory,
-		internalqueue.WithPodInitialBackoffDuration(time.Duration(c.podInitialBackoffSeconds)*time.Second),
-		internalqueue.WithPodMaxBackoffDuration(time.Duration(c.podMaxBackoffSeconds)*time.Second),
-		internalqueue.WithPodNominator(nominator),
-		internalqueue.WithClusterEventMap(c.clusterEventMap),
-		internalqueue.WithPodMaxUnschedulableQDuration(c.podMaxUnschedulableQDuration),
-	)
-
-	// Setup cache debugger.
-	debugger := cachedebugger.New(
-		c.informerFactory.Core().V1().Nodes().Lister(),
-		c.informerFactory.Core().V1().Pods().Lister(),
-		c.schedulerCache,
-		podQueue,
-	)
-	debugger.ListenForSignal(c.StopEverything)
-
-	sched := newScheduler(
-		c.schedulerCache,
-		extenders,
-		internalqueue.MakeNextPodFunc(podQueue),
-		MakeDefaultErrorFunc(c.client, c.informerFactory.Core().V1().Pods().Lister(), podQueue, c.schedulerCache),
-		c.StopEverything,
-		podQueue,
-		profiles,
-		c.client,
-		c.nodeInfoSnapshot,
-		c.percentageOfNodesToScore)
-	return sched, nil
+	return
```

Comment:
remove this

Reply:
Personally would like to keep it as more clearly to me.

Reply:
what is the confusion that you try to clear? it is not typical to explicitly return at the end of a function that returns nothing.

Reply:
ah, you have named return variable. So the best practice is not to use named variables on long functions, can you please explicitly define them in the function and in the return statement.

Reply:
I used to follow the practice: `The names are not mandatory but they can make code shorter and clearer: they're documentation` in https://go.dev/doc/effective_go#named-results. 
Updated then.